### PR TITLE
node, p2p/simulations: fix node.Node AccountsManager leak

### PR DESCRIPTION
**Pull request by janos**
This PR is a second take on fixing the AccountsManager leak on node.Node. The first PR was rejected https://github.com/ethereum/go-ethereum/pull/18505.

Fixes: https://github.com/ethersphere/go-ethereum/issues/1142.

Swarm PR: https://github.com/ethersphere/go-ethereum/pull/1194.

While performing the test described in https://github.com/ethersphere/go-ethereum/issues/1142 issue, a large number of gorutines were created and not terminated. Stack trace shows many goroutines:

```
goroutine 3370 [select]:
github.com/ethereum/go-ethereum/accounts.(*Manager).update(0xc0000b7520)
	/Users/janos/go/src/github.com/ethereum/go-ethereum/accounts/manager.go:95 +0x22a
created by github.com/ethereum/go-ethereum/accounts.NewManager
	/Users/janos/go/src/github.com/ethereum/go-ethereum/accounts/manager.go:68 +0x6f2

goroutine 4481 [select]:
github.com/ethereum/go-ethereum/accounts/keystore.(*watcher).loop(0xc0035ac9a0)
	/Users/janos/go/src/github.com/ethereum/go-ethereum/accounts/keystore/watch.go:94 +0x65b
created by github.com/ethereum/go-ethereum/accounts/keystore.(*watcher).start
	/Users/janos/go/src/github.com/ethereum/go-ethereum/accounts/keystore/watch.go:52 +0xb6
```

Accounts manager is created but not closed for node.Node, leaving large number of goroutines alive. This change closes accounts manager on new method Node.Close, moves the cleaning of ephemeral directory from Stop to Close, and handles the closing of the node in p2p/simulations.Network through p2p/simulations/adapters.SimNode.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -121,6 +121,11 @@ func New(conf *Config) (*Node, error) {
 	}, nil
 }
 
+// Close releases resources acquired in Node constructor New.
+func (n *Node) Close() error {
+	return n.accman.Close()
+}
```
Comment (karalabe) [inline node/node.go]: I would also add a shutdown attempt to make sure everything is cleaned up if someone only calls Close, but not Stop.

```go
func (n *Node) Close() error {
	if err := n.Stop(); err != nil && err != ErrNodeStopped {
		return err
	}
	return n.accman.Close()
}
```

If you want to be truly correct, you could close both first, and only then return the errors, possible returning a combined error if both failed.

```go
func (n *Node) Close() error {
	var errs []error

	// Terminate all subsystems and collect any errors
	if err := n.Stop(); err != nil && err != ErrNodeStopped {
		errs = append(errs, err)
	}
	if err := n.accman.Close(); err != nil {
		errs = append(errs, err)
	}
	// Report any errors that might have occurred
	switch len(errs) {
		case 0: return nil
		case 1: return errs[0]
		default: return fmt.Errorf("%v", errs)
	}
}
```

**Comment by karalabe** [comment]
We also have quite a few `node := makeFullNode(ctx)` invocations in `cmd/geth` that create nodes. To make everything correct, let's also add a `defer node.Close()` after all those code calls.

**Comment by janos** [comment]
Thanks @karalabe for review. I've updated this PR based on your comments.

**Comment by janos** [comment]
Yes, Nodes in node tests are now closed.

**Comment by janos** [comment]
Thanks, @karalabe.

**Pull request merged**